geronimo-openapi-impl/src/main/java/org/apache/geronimo/microprofile/openapi/impl/processor/SchemaProcessor.java

```diff
@@ -145,45 +146,22 @@ public void fillSchema(
                     fillSchema(components, Object.class, items, null);
                     schema.items(items);
                 } else {
-                    ofNullable(from.getAnnotation(Schema.class)).ifPresent(
-                            s -> sets(components, Schema.class.cast(s), schema, null));
+                    Optional<Annotation> annotation = ofNullable(from.getAnnotation(Schema.class));
+                    if (annotation.isPresent()) {
+                        //if providedRef is null and the Schema name is not Empty, we set it as providedRef
+                        Schema schemaAnnotation = Schema.class.cast(annotation.get());
+                        String ref = schemaAnnotation.name().isEmpty() ? null : schemaAnnotation.name();
+                        if (providedRef == null) {
+                            providedRef = ref;
```

Comment:
Shouldnt the ref be resolved (toRef methods) before calling sets?

Reply:
not required, as sets calls again fillSchema and fillSchema is calling then toRef before using the providedRef.